timeutil/schedule.go

```diff
@@ -28,35 +29,49 @@ import (
 	"time"
 )
 
-var validTime = regexp.MustCompile(`^([0-9]|0[0-9]|1[0-9]|2[0-3]):([0-5][0-9])$`)
+// Match 0:00-24:00, where 24:00 means the later end of the day.
+var validTime = regexp.MustCompile(`^(([0-9]|0[0-9]|1[0-9]|2[0-3]):([0-5][0-9]))|(24:00)$`)
 
-type TimeOfDay struct {
+// Clock represents a hour:minute time within a day.
+type Clock struct {
 	Hour   int
 	Minute int
 }
 
-func (t TimeOfDay) String() string {
+func (t Clock) String() string {
 	return fmt.Sprintf("%02d:%02d", t.Hour, t.Minute)
 }
 
 // Sub subtracts `other` TimeOfDay from current and returns duration
-func (t TimeOfDay) Sub(other TimeOfDay) time.Duration {
+func (t Clock) Sub(other Clock) time.Duration {
 	t1 := time.Duration(t.Hour)*time.Hour + time.Duration(t.Minute)*time.Minute
 	t2 := time.Duration(other.Hour)*time.Hour + time.Duration(other.Minute)*time.Minute
 	return t1 - t2
 }
 
 // Add adds given duration and returns a new TimeOfDay
-func (t TimeOfDay) Add(dur time.Duration) TimeOfDay {
+func (t Clock) Add(dur time.Duration) Clock {
 	t1 := time.Duration(t.Hour)*time.Hour + time.Duration(t.Minute)*time.Minute
 	t2 := t1 + dur
-	nt := TimeOfDay{
+	nt := Clock{
 		Hour:   int(t2.Hours()) % 24,
 		Minute: int(t2.Minutes()) % 60,
 	}
 	return nt
 }
 
+// Time generates a time.Time using base for information on year, month, day
+// and with hour and minute set from TimeOfDay
+func (t Clock) Time(base time.Time) time.Time {
+	return time.Date(base.Year(), base.Month(), base.Day(),
+		t.Hour, t.Minute, 0, 0, time.Local)
+}
+
+// isValidTime returns true if given s looks like a valid time specification
+func isValidTime(s string) bool {
```

Comment:
This function can now go away, as the short oneliner below is nice on its own and is only ever used on a single other line.